### DISCUSSION TITLE
Swanctl keeps on disconnecting after a short time of being idle.

### DISCUSSION THREAD

**Discussion by SW1TCH0NY0U** [strongswan/strongswan#2212] [Q&A]
My config:

swanctl:

connections {
privado {
version = 2

proposals = aes192gcm16-aes128gcm16-prfsha256-ecp256-ecp521,aes192-sha256-modp3072,default
local_addrs = "%defaultroute"
remote_addrs = ams-001.vpn.privado.io
vips=0.0.0.0,::

local {
auth = eap-mschapv2
eap_id = *username*
}
remote {
auth = pubkey
# cacerts=/etc/swanctl/NordVPN.pem ### nordvpn cert
id = "%any"
}

children {
privado {
remote_ts = 0.0.0.0/0,::
updown = /etc/vpn
if_id_out = %unique
if_id_in = %unique
rekey_time = 0s
dpd_action = restart
esp_proposals = aes192gcm16-aes128gcm16-prfsha256-ecp256-modp3072,aes192-sha256-ecp256-modp3072,default
start_action = start
}
}
}
}
secrets {
eap-mschapv2 {
id = *username*
secret = *password*
}
}

include conf.d/*.conf

/etc/vpn:

# set charon.install_virtual_ip = no to prevent the daemon from also installing the VIP
sh /etc/route &

set -o nounset
set -o errexit

VTI_IF="Virt0"

case "${PLUTO_VERB}" in
up-client)
ip link add "${VTI_IF}" type xfrm dev eth0 \
if_id "${PLUTO_IF_ID_OUT}"
#ip addr add 10.99.12.1/32 dev "${VTI_IF}"
ip addr add "${PLUTO_MY_SOURCEIP}" dev "${VTI_IF}"
ip link set "${VTI_IF}" mtu 1200
ip link set "${VTI_IF}" up
#ip route add "${PLUTO_MY_SOURCEIP}" dev "${VTI_IF}"
sysctl -w "net.ipv4.conf.${VTI_IF}.disable_policy=1"
ip rule add from "${PLUTO_MY_SOURCEIP}" lookup vpnout
ip route add default dev "${VTI_IF}" table vpnout

;;
down-client)
ip link del "${VTI_IF}"
;;
esac

/etc/route

sleep 4
ip route add 10.0.0.1/32 dev wg0 table vpnout
ip rule add from 10.0.0.1/32 table vpnout
ip rule add to 10.0.0.1/32 table vpnout

constraints:

constraints {

    # Whether to load the plugin. Can also be an integer to increase the
    # priority of this plugin.
    load = no

}

ipsec. secrets:

# This file holds shared secrets or RSA private keys for authentication.

# RSA private key for this host, authenticating it to any other host
# which knows the public part.

*username* : EAP *password*

I reboot, swanctl autostarts but after being idle, it disconnects. I can bring it back up with:

sudo swanctl --load-all
sudo swanctl -i -c privado

How can I stop it from disconnecting?
I've tried everything I can think of!

**Comment by tobiasbrunner** [comment]
Please updated your post and provide logs that show what's actually happening (see #196). (And you could fix the formatting as well.)

**Reply by SW1TCH0NY0U** [reply]
Well, #196 does NOT say how to actually get logs. I looked twice.
Right now it's disconnected so if you could assist in getting logs............

I'm using Ubuntu 22.04!

Thanks.

EDIT:


It says this:

filelog {
        # since 5.7.0 the path to the log file has to be specified in a separate setting if it contains dots,
        # use an arbitrary name without dots for the section instead of the one given here
        charon-debug-log {
                # this setting is required with 5.7.0 and newer if the path contains dots
                path = /var/log/charon_debug.log
                time_format = %a, %Y-%m-%d, %H:%M:%S
                default = 2
                net = 1
                enc = 1
                asn = 1
                job = 1
                ike_name = yes
                append = no
                flush_line = yes
        }
}

Where do I put the "path = /var/log/charon_debug.log" and what is it called/location?

It's still disconnecting. If I reboot, it autostarts and is enabled. But after sometime it cuts out.
Very annoying.

**Reply by tobiasbrunner** [reply]
Is it that difficult to find [this](https://docs.strongswan.org/docs/5.9/config/logging.html)? Anyway, I've added a link to #196.

**Reply by SW1TCH0NY0U** [reply]
What?!

And I quote: "Unlike [charon](https://docs.strongswan.org/docs/5.9/daemons/charon.html), [charon-systemd](https://docs.strongswan.org/docs/5.9/daemons/charon-systemd.html) by default logs to the systemd journal and not to syslog by default. The log levels are configurable in a [separate section](https://docs.strongswan.org/docs/5.9/daemons/charon-systemd.html#_logging) in [strongswan.conf](https://docs.strongswan.org/docs/5.9/config/strongswanConf.html) **which is not described here.**"

So how am I suppose to get logs?!

**Reply by tobiasbrunner** [reply]
Magic?

**Reply by SW1TCH0NY0U** [reply]
pi@raspberrypi:~$ sudo cat /etc/strongswan.conf 
# strongswan.conf - strongSwan configuration file
#
# Refer to the strongswan.conf(5) manpage for details
#
# Configuration changes should be made in the included files

charon {
load_modular = no
threads = 16
force_keepalive = yes
keep_alive = 30
plugins {
include strongswan.d/charon/*.conf
}
start-scripts {
load-all = /usr/sbin/swanctl --load-all
}
}

charon {

  # two defined file loggers
  filelog {
    charon {
      # path to the log file, specify this as section name in versions prior to 5.7.0
      path = /var/log/charon.log
      # add a timestamp prefix
      time_format = %b %e %T
      # prepend connection name, simplifies grepping
      ike_name = yes
      # overwrite existing files
      append = no
      # increase default loglevel for all daemon subsystems
      default = 2
      # flush each line to disk
      flush_line = yes
    }
    stderr {
      # more detailed loglevel for a specific subsystem, overriding the
      # default loglevel.
      ike = 2
      knl = 3
    }
  }

  # and two loggers using syslog
  syslog {
    # prefix for each log message
    identifier = charon-custom
    # use default settings to log to the LOG_DAEMON facility
    daemon {
    }
    # very minimalistic IKE auditing logs to LOG_AUTHPRIV
    auth {
      default = -1
      ike = 0
    }
  }
 # ...
}

include strongswan.d/*.conf

Now bring it up and the logs will be there?

**Reply by SW1TCH0NY0U** [reply]
pi@raspberrypi:~$ vpn-on
no files found matching '/etc/swanctl/conf.d/*.conf'
loaded eap secret 'eap-mschapv2'
no authorities found, 0 unloaded
no pools found, 0 unloaded
loaded connection 'privado'
successfully loaded 1 connections, 0 unloaded
[IKE] establishing CHILD_SA privado{4}
[ENC] generating CREATE_CHILD_SA request 0 [ SA No KE TSi TSr ]
[NET] sending packet: from 192.168.1.145[4500] to 91.148.224.10[4500] (572 bytes)
[NET] received packet: from 91.148.224.10[4500] to 192.168.1.145[4500] (204 bytes)
[ENC] parsed CREATE_CHILD_SA response 0 [ SA No TSi TSr ]
[CFG] selected proposal: ESP:AES_CBC_128/HMAC_SHA1_96/NO_EXT_SEQ
[IKE] ignoring KE exchange, agreed on a non-PFS proposal
[IKE] CHILD_SA privado{4} established with SPIs c988ec07_i c5744665_o and TS 172.20.152.59/32 === 0.0.0.0/0
[CHD] updown: net.ipv4.conf.Virt0.disable_policy = 1
[CHD] updown: RTNETLINK answers: File exists
initiate completed successfully

Now it'll just disconnect after being idle!

**Reply by SW1TCH0NY0U** [reply]
/var/log/charon.log

Is empty.............

**Reply by SW1TCH0NY0U** [reply]
Did you even look at my settings to make sure they're okay?

**Reply by tobiasbrunner** [reply]
Sorry, but why should I or anybody else do your work?

**Reply by SW1TCH0NY0U** [reply]
What?!

I'm *not* asking you to do my "work" - I'm asking you if you can help me figure out why it's disconnecting.

**Reply by SW1TCH0NY0U** [reply]
Where do you want the log uploaded to?
It's too big for Github and too big for debian's paste.

**Reply by SW1TCH0NY0U** [reply]
Where do I put this?

"dpdaction = restart"

In /etc/swanctl/swanctl.conf OR /etc/strongwan.conf?

**Reply by SW1TCH0NY0U** [reply]
loading connection 'privado' failed: unknown option: closeaction, config discarded

**Reply by SW1TCH0NY0U** [reply]
Added this:

# strongswan.conf - strongSwan configuration file
#
# Refer to the strongswan.conf(5) manpage for details
#
# Configuration changes should be made in the included files

charon {
load_modular = no
threads = 16
force_keepalive = yes
keep_alive = 30
**dpdaction = restart
closeaction = restart**
plugins {
include strongswan.d/charon/*.conf
}
start-scripts {
load-all = /usr/sbin/swanctl --load-all
}
}

charon {

  # two defined file loggers
  filelog {
    charon {
      # path to the log file, specify this as section name in versions prior to 5.7.0
      path = /var/log/charon.log
      # add a timestamp prefix
      time_format = %b %e %T
      # prepend connection name, simplifies grepping
      ike_name = yes
      # overwrite existing files
      append = no
      # increase default loglevel for all daemon subsystems
      default = 2
      # flush each line to disk
      flush_line = yes
    }
    stderr {
      # more detailed loglevel for a specific subsystem, overriding the
      # default loglevel.
      ike = 2
      knl = 3
    }
  }

  # and two loggers using syslog
  syslog {
    # prefix for each log message
    identifier = charon-custom
    # use default settings to log to the LOG_DAEMON facility
    daemon {
    }
    # very minimalistic IKE auditing logs to LOG_AUTHPRIV
    auth {
      default = -1
      ike = 0
    }
  }
 # ...
}

include strongswan.d/*.conf

**Reply by SW1TCH0NY0U** [reply]
Seems to be "up" and I'll idle for 10 minutes or so to see if it disconnects.
Sorry if I came off the wrong way, been having this issue for a while hence the thread on Github. :dancers:

**Comment by SW1TCH0NY0U** [comment]
Seems to be staying up. No disconnections yet. Thanks 💯😊

**Comment by SW1TCH0NY0U** [comment]
Hi, it's still disconnecting. I'll try and post logs.
